### PR TITLE
Fix Field MemberName init value of vmtarget

### PR DESCRIPTION
**Pull request by fengxue-IS**
Fixes: #11938 

Signed-off-by: Jack Lu <Jack.S.Lu@ibm.com>

### CODE REVIEW

**Comment by fengxue-IS** [comment]
local testing passed, reruning personal build to confirm no test regression

**Review by babsingh — APPROVED** [review]

**Comment by babsingh** [comment]
LGTM. @tajila Can you please review, test and merge this PR?

**Comment by babsingh** [comment]
Note: This PR will also fix https://github.com/eclipse-openj9/openj9/issues/11937.

File: runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp [hunk 1]
```diff
@@ -96,8 +96,17 @@ initImpl(J9VMThread *currentThread, j9object_t membernameObject, j9object_t refO
...
+
+		if (J9_ARE_ANY_BITS_SET(romField->modifiers, J9AccStatic)) {
+			offset = fieldID->offset | J9_SUN_STATIC_FIELD_OFFSET_TAG;
+			if (J9_ARE_ANY_BITS_SET(romField->modifiers, J9AccFinal)) {
+				offset |= J9_SUN_FINAL_FIELD_OFFSET_TAG;
+			}
+		}
 		vmindex = (jlong)fieldID;
-		target = (jlong)fieldID->field;
+		target = (jlong)offset;
```
Comment (tajila) [inline runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp:109]: the comment above needs to be updated

File: runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp [hunk 2]
```diff
@@ -96,8 +96,17 @@ initImpl(J9VMThread *currentThread, j9object_t membernameObject, j9object_t refO
 
 	if (refClass == J9VMJAVALANGREFLECTFIELD(vm)) {
 		J9JNIFieldID *fieldID = vm->reflectFunctions.idFromFieldObject(currentThread, NULL, refObject);
+		J9ROMFieldShape *romField = fieldID->field;
+		UDATA offset = fieldID->offset;
+
+		if (J9_ARE_ANY_BITS_SET(romField->modifiers, J9AccStatic)) {
+			offset = fieldID->offset | J9_SUN_STATIC_FIELD_OFFSET_TAG;
```
Comment (tajila) [inline runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp:103]: can this just be `offset |= J9_SUN_STATIC_FIELD_OFFSET_TAG;`

**Comment by tajila** [comment]
General question: what was `target` set to the romfieldshape in the first place?

File: runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp (hunk 2 above)
Comment (fengxue-IS) [reply runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp]: fixed

File: runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp (hunk 1 above)
Comment (fengxue-IS) [reply runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp:109]: fixed

**Comment by fengxue-IS** [comment]
> General question: what was `target` set to the romfieldshape in the first place?

I think originally we were only using `target` to check modifiers so we stored the romfieldshape similar to how method pointer is stored, and the offset is computed with `JNIFieldID->offset`. Later updated the offset compute to include the static/final tag bits and changed `target` to store offset so we don't have to recompute every time. but forgot to update it here?

**Review by tajila — APPROVED** [review]

**Comment by tajila** [comment]
jenkins test sanity xlinuxojdk292 jdk16 depends ibmruntimes/openj9-openjdk-jdk16#29

**Comment by tajila** [comment]
@fengxue-IS can you double check that all the test failures are known issues

**Comment by fengxue-IS** [comment]
@tajila confirmed all test failures are known issues

**Pull request merged**